components/brave_sync/brave_sync_prefs_mem_store.h

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SYNC_BRAVE_SYNC_PREFS_MEM_STORE_H_
+#define BRAVE_COMPONENTS_BRAVE_SYNC_BRAVE_SYNC_PREFS_MEM_STORE_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/callback.h"
+#include "base/macros.h"
+#include "base/values.h"
+#include "brave/components/brave_sync/brave_sync_prefs_base.h"
+
+namespace base {
+class Time;
+}
+
+namespace brave_sync {
+
+class Settings;
+class SyncDevices;
+
+namespace prefs {
+
+class PrefsMemStore : public PrefsBase {
```

Comment:
why do we need `PrefsBase`? `PrefsMemStore` can be a subclass of `Prefs` and you can still do the switch.

Reply:
Yes, it was not required, but since `PrefsMemStore` would override all methods of `Prefs`, I did as it is there now.